src/constants/dom-events.ts

```diff
@@ -62,6 +62,41 @@ export interface DomEvent {
 
 export type DomEventName = keyof typeof domEvents
 
+export const ignorableKeyModifiers = ['stop', 'prevent', 'self', 'exact', 'prevent', 'capture']
+export const systemKeyModifiers = ['ctrl', 'shift', 'alt', 'meta'] as const
+export const mouseKeyModifiers = ['left', 'middle', 'right'] as const
+
+export const keyCodesByKeyName = {
+  backspace: 8,
+  tab: 9,
+  enter: 13,
+  esc: 27,
+  space: 32,
+  pageup: 33,
+  pagedown: 34,
+  end: 35,
+  home: 36,
+  left: 37,
+  up: 38,
+  right: 39,
+  down: 40,
+  insert: 45,
+  delete: 46
+} as const
+
+export type KeyName = keyof typeof keyCodesByKeyName
+export type Modifier =
+  | typeof systemKeyModifiers[number]
+  | typeof mouseKeyModifiers[number]
+
+export type DomEventNameWithModifier =
+  | DomEventName
+  | `${DomEventName}.${typeof systemKeyModifiers[number]}`
+  | `click.${typeof mouseKeyModifiers[number]}`
+  | `click.${typeof systemKeyModifiers[number]}.${typeof mouseKeyModifiers[number]}`
+  | `${'keydown' | 'keyup'}.${keyof typeof keyCodesByKeyName}`
+  | `${'keydown' | 'keyup'}.${typeof systemKeyModifiers[number]}.${keyof typeof keyCodesByKeyName}`
```

Comment:
these 3 lines can have even more modifiers no?
Like: `click.ctrl.shift.middle`?

Reply:
I have not actually combined them like that before, but I guess you can? Would need to verify. 

There is also a bunch of other ones apparently:

```
<!-- Vue 3 Key Modifier on v-on -->
<input @keyup.page-down="nextPage">

<!-- Matches both q and Q -->
<input @keypress.q="quit">
```

Reply:
I saw that there's `page-down` https://v3.vuejs.org/guide/events.html#key-modifiers (second example) therefore `page-up` must be a thing too. That has to be an alternative to `pagedown`, `pageup` which is already a thing in vtu
I just figured I won't bother as it's out of scope of this PR but if you can link to a list of modifiers I could add them.

> I have not actually combined them like that before, but I guess you can? Would need to verify.

I imagine it's a thing and probably you could chain as many you want.
This might be where it's actually processed: https://github.com/vuejs/vue-next/blob/8ed3ed6c27/packages/runtime-dom/src/directives/vOn.ts#L37

I would say so long we cover the most common ones for everything else `string` is still accepted

What do you think?

Reply:
@nandi95 Yeah I think this would be enough: we'll never be able to cover all cases, but the point is to just have a bit of help for the most common ones, and fall back to string for the rest.

Reply:
agreed, good enough is good enough sometimes,